src/near-safe.ts

```diff
@@ -478,14 +524,36 @@ export class NearSafe {
         });
         const opHash = await this.opHash(chainId, userOp);
         return {
-          payload: toPayload(opHash),
           evmMessage: JSON.stringify(userOp),
-          hash: await this.opHash(chainId, userOp),
+          // We don't need both here.
+          payload: toPayload(opHash),
+          hash: opHash,
         };
       }
     }
   }
 
+  encodeForSafe(from: string): boolean {
+    const fromAddress = getAddress(from);
+    if (![this.address, this.mpcAddress].includes(fromAddress)) {
+      throw new Error(`Unexpected from address ${from}`);
+    }
+    return this.address === fromAddress;
+  }
+
+  async eoaEncode(request: SignRequestData): Promise<EncodedSignRequest> {
+    // TODO: Adapt nearAdapter.encodeSignRequest to return correct details.
```

Comment:
Fix is here: Will be released soon.

https://github.com/BitteProtocol/near-ca/pull/142